src/main/jbake/content/blog/2017/03/24/distributed-tracing-with-camel.adoc

```diff
@@ -0,0 +1,267 @@
+    </dependency>
+```
+
+and any other dependencies specific to the OpenTracing compliant Tracer implementation being used. Then simply annotate the application using `CamelOpenTracing`:
+
+```java
+@SpringBootApplication
+@CamelOpenTracing
+public class MyApplication {
+    public static void main(String[] args) {
+	// Location for initializing the Tracer if not implicitly loaded
+        SpringApplication.run(MyApplication.class, args);
+    }
+}
+```
+
+See the https://github.com/apache/camel/tree/master/examples/camel-example-opentracing/service1[camel-opentracing example service1 application] for an example.
+
+==== Java Agent
+
+This approach uses a `javaagent` to handle the initialization, so does not require any code/configuration changes. The dependencies required are:
+
+```xml
+    <dependency>
+      <groupId>org.apache.camel</groupId>
+      <artifactId>camel-opentracing</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.opentracing.contrib</groupId>
+      <artifactId>opentracing-agent</artifactId>
+      <scope>provided</scope>
+    </dependency>
+```
+
+Changes are also required in the `pom.xml` to download the `opentracing-agent.jar` and then execute the application supplying the java agent on the command line.
+
+See the https://github.com/apache/camel/tree/master/examples/camel-example-opentracing/service2[camel-opentracing example service2 application] for an example.
+
+=== Instrumenting Some Examples
+
+This section will show how two examples can be instrumented using two different OpenTracing compliant Tracer implementations.
+
+The first is a new example introduced to demonstrate the three ways a Camel application can be instrumented with OpenTracing, and will be updated to use the http://zipkin.io/[Zipkin] project.
+
+The second shows how an existing example (Loan Broker JMS) can be instrumented using the http://www.hawkular.org/hawkular-apm/[Hawkular APM] project.
+
+'''
+
+==== Camel OpenTracing Example
+
+This section will show how the new example, which uses a simple _logging_ tracer implementation by default, can be changed to use Zipkin.
+
+First edit each of the client/service1/service2 `pom.xml` files to:
+
+* Add properties
+
+```xml
+  <properties>
+    ...
+    <version.brave.opentracing>0.18.3</version.brave.opentracing>
+    <version.zipkin.reporter>0.6.12</version.zipkin.reporter>
+  </properties>
+```
+
+* Replace the `camel-example-opentracing-loggingtracer` dependency with:
+
+```xml
+    <dependency>
+      <groupId>io.opentracing.brave</groupId>
+      <artifactId>brave-opentracing</artifactId>
+      <version>${version.brave.opentracing}</version>
+    </dependency>
+    <dependency>
+      <groupId>io.zipkin.reporter</groupId>
+      <artifactId>zipkin-sender-urlconnection</artifactId>
+      <version>${version.zipkin.reporter}</version>
+    </dependency>
+```
+
+* Common code for ClientApplication Service1Application and ServiceApplication
+
+```java
```

Comment:
I think it could make sense to have a single `+` on the previous line so that the example is indended to the level of the bulled point and is not left-aligned.